crates/jstzd/src/task/inner_octez_node.rs

```diff
@@ -0,0 +1,79 @@
+#![allow(unused)]
+use std::{fs::File, path::PathBuf, process::Stdio};
+
+use tokio::process::{Child, Command};
+
+use anyhow::{anyhow, Result};
+
+fn path_or_default<'a>(path: Option<&'a PathBuf>, default: &'a str) -> &'a str {
+    path.and_then(|bin| bin.to_str()).unwrap_or(default)
+}
+
+async fn run_command(command: &mut Command) -> Result<()> {
+    let output = command.output().await?;
+
+    if !output.status.success() {
+        return Err(anyhow!(
+            "Command {:?} failed:\n {}",
+            command,
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    Ok(())
+}
+
+pub struct InnerOctezNode {
+    /// Path to the octez-node binary
+    /// If None, the binary will inside PATH will be used
+    pub octez_node_bin: Option<PathBuf>,
+    /// Path to the octez-node directory
+    pub octez_node_dir: PathBuf,
+}
+
+impl InnerOctezNode {
+    fn command(&self) -> Command {
+        Command::new(path_or_default(self.octez_node_bin.as_ref(), "octez-node"))
+    }
+
+    pub async fn config_init(
+        &self,
+        network: &str,
+        http_endpoint: &str,
+        rpc_endpoint: &str,
+        num_connections: u32,
+    ) -> Result<()> {
+        run_command(self.command().args([
+            "config",
+            "init",
+            "--network",
+            network,
+            "--data-dir",
+            self.octez_node_dir.to_str().expect("Invalid path"),
+            "--net-addr",
+            http_endpoint,
+            "--rpc-addr",
+            rpc_endpoint,
+            "--connections",
+            num_connections.to_string().as_str(),
+        ]))
+        .await
+    }
+
+    pub async fn run(&self, log_file: &File, options: &[&str]) -> Result<Child> {
+        let mut command = self.command();
+
+        command
+            .args([
+                "run",
+                "--data-dir",
+                self.octez_node_dir.to_str().expect("Invalid path"),
+                "--singleprocess",
```

Comment:
I think we should just run validation in a separate process to avoid unresponsive node hiccups.